app/models/manageiq/providers/infra_manager.rb

```diff
@@ -7,6 +7,18 @@ class InfraManager < BaseManager
 
     include AvailabilityMixin
 
+    has_many :host_hardwares,         :through => :hosts, :source => :hardware
+    has_many :host_operating_systems, :through => :hosts, :source => :operating_system
+    has_many :host_storages,          :through => :hosts
+    has_many :host_switches,          :through => :hosts
+    has_many :host_networks,          :through => :hosts, :source => :networks
+    has_many :snapshots,              :through => :vms_and_templates
+    has_many :operating_systems,      :through => :vms_and_templates
+    has_many :switches,               :through => :hardwares
+    has_many :lans,                   :through => :hardwares
```

Comment:
If anything I think that makes this more confusing since the common pattern for this is [ref](http://guides.rubyonrails.org/association_basics.html#the-has-many-through-association)
```
has_many :host_switches
has_many :switches, :through => :host_switches
```

If we have `:host_switches` then we should get switches and lans through that